homeassistant/components/sonos/speaker.py

```diff
@@ -126,6 +131,7 @@ def __init__(
         self._last_event_cache: dict[str, Any] = {}
         self.activity_stats: ActivityStatistics = ActivityStatistics(self.zone_name)
         self.event_stats: EventStatistics = EventStatistics(self.zone_name)
+        self._in_cooldown: bool = False
```

Comment:
Instead of storing True/False and then registering a listener to set it to False again, I would store a timestamp until when a connection shouldn't be retried.